pkg/controllers/clusterresourceplacementeviction/controller.go

```diff
@@ -0,0 +1,257 @@
+		maxUnavailable, _ := intstr.GetScaledValueFromIntOrPercent(db.Spec.MaxUnavailable, desiredBindings, true)
+		unavailableBindings := len(bindings) - availableBindings
+		disruptionsAllowed = maxUnavailable - unavailableBindings
+	}
+	if db.Spec.MinAvailable != nil {
+		minAvailable, _ := intstr.GetScaledValueFromIntOrPercent(db.Spec.MinAvailable, desiredBindings, true)
+		disruptionsAllowed = availableBindings - minAvailable
+	}
+	if disruptionsAllowed < 0 {
+		disruptionsAllowed = 0
+	}
+	return disruptionsAllowed > 0, disruptionsAllowed, availableBindings
+}
+
+// markEvictionValid sets the valid condition as true in eviction status.
+func markEvictionValid(eviction *placementv1alpha1.ClusterResourcePlacementEviction) {
+	cond := metav1.Condition{
+		Type:               string(placementv1alpha1.PlacementEvictionConditionTypeValid),
+		Status:             metav1.ConditionTrue,
+		ObservedGeneration: eviction.Generation,
+		Reason:             reasonClusterResourcePlacementEvictionValid,
+		Message:            evictionValid,
+	}
+	eviction.SetConditions(cond)
+}
+
+// markEvictionInvalid sets the valid condition as false in eviction status.
+func markEvictionInvalid(eviction *placementv1alpha1.ClusterResourcePlacementEviction, message string) {
+	cond := metav1.Condition{
+		Type:               string(placementv1alpha1.PlacementEvictionConditionTypeValid),
+		Status:             metav1.ConditionFalse,
+		ObservedGeneration: eviction.Generation,
+		Reason:             reasonClusterResourcePlacementEvictionInvalid,
+		Message:            message,
+	}
+	eviction.SetConditions(cond)
+}
+
+// markEvictionExecuted sets the executed condition as true in eviction status.
+func markEvictionExecuted(eviction *placementv1alpha1.ClusterResourcePlacementEviction, message string) {
+	cond := metav1.Condition{
+		Type:               string(placementv1alpha1.PlacementEvictionConditionTypeExecuted),
+		Status:             metav1.ConditionTrue,
+		ObservedGeneration: eviction.Generation,
+		Reason:             reasonClusterResourcePlacementEvictionExecuted,
+		Message:            message,
+	}
+	eviction.SetConditions(cond)
+}
+
+// markEvictionNotExecuted sets the executed condition as false in eviction status.
+func markEvictionNotExecuted(eviction *placementv1alpha1.ClusterResourcePlacementEviction, message string) {
+	cond := metav1.Condition{
+		Type:               string(placementv1alpha1.PlacementEvictionConditionTypeExecuted),
+		Status:             metav1.ConditionFalse,
+		ObservedGeneration: eviction.Generation,
+		Reason:             reasonClusterResourcePlacementEvictionNotExecuted,
+		Message:            message,
+	}
+	eviction.SetConditions(cond)
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *Reconciler) SetupWithManager(mgr runtime.Manager) error {
+	return runtime.NewControllerManagedBy(mgr).
+		WithOptions(ctrl.Options{MaxConcurrentReconciles: 1}). // set the max number of concurrent reconciles
```

Comment:
A nit: we might need to add a note here to explain why we have to use 1 max. concurrent reconciliations for now.